src/dataloader.ts

```diff
@@ -48,6 +50,7 @@ export interface LoaderOptions {
 export class LoaderResolver {
   private readonly root: string;
   private readonly interpreters: Map<string, string[]>;
+  lastModified: Map<string, number>;
```

Comment:
A risk with this design is that you try to access the `lastModified` time from the map before it’s been initialized (because you haven’t called `getFileHash` yet). This isn’t a problem with the other sets/maps that `getResolvers` returns because those maps are fully-initialized when `getResolvers` returns, and subsequently never mutated. But it is a problem here because the map is mutable, and visible before the map is populated.

I would therefore instead prefer that the `LoaderResolver` class expose a method `getFileLastModified` that returns the (indirect) last-modified time of the given file. I also probably wouldn’t cache this information. (Note that we store `mtimeMs` in the `fileInfoCache` used by `getFileInfo`, but that’s just used to invalidate the cache — we check the current `mtimeMs` every time we read from `fileInfoCache`. Calling `statSync` should be fast.)